src/common/CommonUtils.ts

```diff
@@ -20,7 +20,7 @@ export class CommonUtils {
         return Promise.resolve();
     }
 
-    public static delay(ms: number): Promise<void> {
+    public static async delay(ms: number): Promise<void> {
```

Comment:
LOL I find this a little amusing after your last epic code clean-up. 😉 Nice catch! 😀

Reply:
yes, looks like I'm the one who's been contributing to the problem 😆